### PR TITLE
Implement reduction avg op end to end

### PR DESCRIPTION
**Pull request by dgolubovicTT**
Solves #243. Implement avg reduction through TTIR, TTNN dialects and generate flat buffers. Add mapping between dialects for avg op.Add simple_avg.mlir test.

### CODE REVIEW

File: test/ttmlir/Dialect/TTNN/simple_avg.mlir [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// RUN: ttmlir-opt --ttir-layout --ttnn-open-device --convert-ttir-to-ttnn %s | FileCheck %s
+#any_device = #tt.operand_constraint<dram|l1|tile|any_device|any_device_tile>
+module attributes {tt.system_desc = #tt.system_desc<[{arch = <wormhole_b0>, grid = 8x8, l1_size = 1048576, num_dram_channels = 12, dram_channel_size = 1048576, noc_l1_address_align_bytes = 16, pcie_address_align_bytes = 32, noc_dram_address_align_bytes = 32}], [0], [<pcie|host_mmio>], [<0, 0, 0, 0>]>} {
+  func.func @forward(%arg0: tensor<512x1024xbf16>) -> tensor<512x32xbf16> {
+    // CHECK: %[[C:.*]] = "ttnn.open_device"[[C:.*]]
+    // CHECK: %[[C:.*]] = "ttnn.full"[[C:.*]]
+    %0 = tensor.empty() : tensor<512x32xbf16>
+    // CHECK: %[[C:.*]] = "ttnn.to_memory_config"[[C:.*]]
+    // CHECK: %[[C:.*]] = "ttnn.sum"[[C:.*]]
```
Comment (derdeljanTT) [inline test/ttmlir/Dialect/TTNN/simple_avg.mlir:9]: Shouldn't this check be `ttnn.avg`?

Comment (dgolubovicTT) [reply test/ttmlir/Dialect/TTNN/simple_avg.mlir:9]: Right, my bad.

**Review by rpavlovicTT — APPROVED** [review]

**Comment by jnie-TT** [comment]
@dgolubovicTT did all CI jobs pass for this? When building locally I'm seeing this error:
```bash
/localdev/jnie/tt-mlir/runtime/lib/ttnn/program.cpp:210:34: error: no member named 'avg' in namespace 'ttnn'
  210 |     tensorPool.push_back(::ttnn::avg(in, dim_arg, op->keep_dim()));
      |                          ~~~~~~~~^
1 error generated.
[166/185] Building CXX object python/CMakeFiles/TTMLIRPythonModules.extension._ttmlir.dso.dir/TTModule.cpp.o
ninja: build stopped: subcommand failed.
```

**Comment by dgolubovicTT** [comment]
Seems not. I will revert this one until I solve it. Sorry for that one, still getting used to github pipelines.

**Pull request merged**